File: src/Neuralm.Application.Messages/Requests/EndTrainingSessionRequest.cs
﻿using System;

namespace Neuralm.Application.Messages.Requests
{
    /// <summary>
    /// Represents the <see cref="EndTrainingSessionRequest"/> class.
    /// </summary>
    public class EndTrainingSessionRequest : Request
    {
        /// <summary>
        /// Gets the training session id.
        /// </summary>
        public Guid TrainingSessionId { get; }

        /// <summary>
        /// Initializes an instance of the <see cref="EndTrainingSessionRequest"/> class.
        /// </summary>
        /// <param name="trainingSessionId"></param>
        public EndTrainingSessionRequest(Guid trainingSessionId)
        {
            TrainingSessionId = trainingSessionId;
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/requests/EndTrainingSessionRequest.ts
import Request from './Request';

/**
 * Represents the EndTrainingSessionRequest class.
 */
export default class EndTrainingSessionRequest extends Request {
  public trainingSessionId: string;

  /**
   * Initializes the EndTrainingSessionRequest class.
   * @param userId The user id.
   * @param trainingRoomId The training room id.
   */
  constructor(trainingSessionId: string) {
    super();
    this.trainingSessionId = trainingSessionId;
  }
}
